src/app/GitUI/CommandsDialogs/BrowseDialog/FormGoToCommit.cs

```diff
@@ -6,6 +6,8 @@ namespace GitUI.CommandsDialogs.BrowseDialog
 {
     public sealed partial class FormGoToCommit : GitModuleForm
     {
+        private const int _maxDropDownCount = 20_000;
```

Comment:
This feels like a lot.... Why this many and, say, not 1,000 or 3,000?

Reply:
> This feels like a lot.... Why this many and, say, not 1,000 or 3,000?

It is a restriction. That's why I chose a larger number with which the UI is blocked shortly enough.
I even was afraid someone would request a button in order to load all refs.
(Furthermore, the ComboBox dropdown is unable to list more than ~ 35,000 entries without weird scrollbar behavior.)